### DISCUSSION TITLE
Server Errors/Warnings

### DISCUSSION THREAD

**Discussion by lvca** [ArcadeData/arcadedb#1133] [Ideas]
I'd like to have a space where to report server errors and warnings, easily accessible by the user. 

In the server the API could be something like this:

```java
reportError( SEVERITY severity, String databaseName, String component, String message )
```

The server could save those messages in a special `jsonl` (one json per line) file.

Some examples of error/warnings on top of my mind:
- Disk space is almost full
- JVM Used Heap > 80%
- Many attempts of login in error (in the message there should be user name and source IP)
- Queries that repeatedly scan big buckets without using an index
- Queries that repeatedly filter by fields that are not indexed with the suggestion about how to create a new index

**Comment by vic0824** [comment]
It could also be useful to report large GC pauses, if possible.

**Reply by lvca** [reply]
That's a great idea. I'm going to include hotspot safepoint stats in the metrics and create a new ServerMontitor component that checks this (and other metrics) reporting when there is a spike.

**Reply by lvca** [reply]
Implemented.

**Comment by vic0824** [comment]
The sever event JSONL file contains some incorrectly formatted characters:
`{"time":"2023-06-30 08:55:04.269","type":"INFO","component":"Server","db":null,"message":"ArcadeDB Server started in \u0027development\u0027 mode (CPUs\u003d8 MAXRAM\u003d3.93GB)"}`

**Reply by lvca** [reply]
It's just how it's encode, but then on the browser looks fine:
![image](https://github.com/ArcadeData/arcadedb/assets/312606/59c1ea07-0ed3-4c7a-986c-12bd40769b2c)
